### PR TITLE
check for read-only sample fields when doing bulk upsert

### PR DESCRIPTION
**Pull request by shriramshankar**

### CODE REVIEW

File: worker/jobs/bulkUpsertSamplesJob.js [hunk 1]
```diff
@@ -38,9 +39,11 @@ module.exports = (job, done) => {
 
   let bulkUpsertPromise;
   if (featureToggles.isFeatureEnabled('enableRedisSampleStore')) {
```
Comment (iamigo) [inline worker/jobs/bulkUpsertSamplesJob.js]: Maybe this is a little cleaner... ?

```
const sampleModel = featureToggles.isFeatureEnabled('enableRedisSampleStore')) ?
    cacheSampleModel : helper.model;
sampleModel.bulkUpsertByName(samples, userName, readOnlyFields);
```

File: tests/cache/models/samples/upsertBulk.js [hunk 2]
```diff
@@ -290,6 +290,62 @@ describe('api::redisEnabled::POST::bulkUpsert ' + path, () => {
...
+    .end((err, res) => {
+      if (err) {
+        return done(err);
+      }
+      expect(res.body.status).to.equal('OK');
+      return done();
+    });
+  });
+
+  it('samples with read only fields in them should not be upserted', (done) => {
```
Comment (iamigo) [inline tests/cache/models/samples/upsertBulk.js]: Do we reject the whole batch if even one sample has a read-only field?

File: api/v1/controllers/samples.js [hunk 3]
```diff
@@ -216,14 +216,14 @@ module.exports = {
         wrappedBulkUpsertData.upsertData = value;
         wrappedBulkUpsertData.userName = userName;
         wrappedBulkUpsertData.reqStartTime = reqStartTime;
-
+        wrappedBulkUpsertData.readOnlyFields = helper.readOnlyFields;
         const j = jobWrapper.createJob(jobType.BULKUPSERTSAMPLES,
           wrappedBulkUpsertData, req);
       } else {
         const bulkUpsertPromise =
         featureToggles.isFeatureEnabled(sampleStoreConstants.featureName) ?
```
Comment (iamigo) [inline api/v1/controllers/samples.js]: Maybe this is a little cleaner... ?

```
const sampleModel = featureToggles.isFeatureEnabled(sampleStoreConstants.featureName) ?
    redisModelSample : helper.model;
sampleModel.bulkUpsertByName(value, userName, readOnlyFields);
```

File: tests/cache/models/samples/upsertBulk.js (hunk 2 above)
Comment (shriramshankar) [inline tests/cache/models/samples/upsertBulk.js]: No, just rejecting the ones with the read-only field in them. I can update the test to show that.

File: tests/cache/models/samples/upsertBulk.js [hunk 4]
```diff
@@ -290,6 +286,67 @@ describe('api::redisEnabled::POST::bulkUpsert ' + path, () => {
...
+       */
+      setTimeout(() => {
+        api.get('/v1/samples?name=' +
+        `${tu.namePrefix}Subject|${tu.namePrefix}Aspect*`)
+        .end((err, res) => {
+          if (err) {
+            done(err);
+          }
+          expect(res.body).to.have.length(2);
+          expect(res.body[0].value).to.not.equal('10');
```
Comment (annyhe) [inline tests/cache/models/samples/upsertBulk.js]: what is the value of the res.body[0].value?

Comment (annyhe) [inline tests/cache/models/samples/upsertBulk.js]: Also, should we check for errorCount?

Comment (shriramshankar) [inline tests/cache/models/samples/upsertBulk.js]: There is no errorCount field on the GET samples endpoint. The value of res.body[0].value is set by the test that was run before this.

**Pull request merged**